core/src/main/java/io/questdb/griffin/engine/functions/conditional/NullIfIFunctionFactory.java

```diff
@@ -70,7 +70,10 @@ public Function getArg() {
         @Override
         public int getInt(Record rec) {
             final int val = value.getInt(rec);
-            return val != Numbers.INT_NaN ? val : replacement;
+            if (val == Numbers.INT_NaN) {
```

Comment:
minor: I think it's ok without this `if`

Reply:
Yes, you're right. Removed in fe3a9cd2fbf47e646417a97506cd729d084f8fc1